models/enterprise_sonic/l2_acls/sonic_l2_acls.yml

```diff
@@ -0,0 +1,206 @@
+            suboptions:
+              sequence_num:
+                description:
+                  - Specifies the sequence number of the rule.
+                  - The range is from 1 to 65535.
+                type: int
+                required: true
+              action:
+                description:
+                  - Specifies the action taken on the matched Ethernet frame.
+                type: str
+                choices:
+                  - deny
+                  - discard
+                  - do-not-nat
+                  - permit
+                  - transit
+              source:
+                description:
+                  - Specifies the source of the Ethernet frame.
+                  - I(address) and I(address_mask) are required together.
+                  - I(any), I(host) and I(address) are mutually exclusive.
+                type: dict
+                mutually_exclusive: [[any, host, address]]
+                required_together: [[address, address_mask]]
+                suboptions:
+                  any:
+                    description:
+                      - Match any source MAC address.
+                    type: bool
+                  host:
+                    description:
+                      - MAC address of a single source host.
+                    type: str
+                  address:
+                    description:
+                      - Source MAC address.
+                    type: str
+                  address_mask:
+                    description:
+                      - Source MAC address mask.
+                    type: str
+              destination:
+                description:
+                  - Specifies the destination of the Ethernet frame.
+                  - I(address) and I(address_mask) are required together.
+                  - I(any), I(host) and I(address) are mutually exclusive.
+                type: dict
+                mutually_exclusive: [[any, host, address]]
+                required_together: [[address, address_mask]]
+                suboptions:
+                  any:
+                    description:
+                      - Match any destination MAC address.
+                    type: bool
+                  host:
+                    description:
+                      - MAC address of a single destination host.
+                    type: str
+                  address:
+                    description:
+                      - Destination MAC address.
+                    type: str
+                  address_mask:
+                    description:
+                      - Destination MAC address mask.
+                    type: str
+              ethertype:
+                description:
+                  - Specifies the EtherType of the Ethernet frame.
+                  - Only one suboption can be specified for ethertype in a rule.
+                type: dict
+                mutually_exclusive: [[value, arp, ipv4, ipv6]]
```

Comment:
Should "required_one_of" also be specified here? (What happens if none of the types is specified?)

Reply:
Since `ethertype` is an optional attribute, not specifying any of its sub-options is acceptable.

Reply:
In `ethertype`, all the options are 'mutually_exclusive' and thus only one option can be specified. If no options are specified, then `ethertype` will be None. In this case no option can be made 'required' and 'required_one_of' is also not necessary.

In case of `source` or `destination` of sonic_l3_acls, `any` , `host` and `prefix` are mutually exclusive while `port_number` is valid with any option. But `port_number` alone is not a valid option, hence rest of the options are included in 'required_one_of' to ensure that `port_number` is not specified as a standalone option.

Reply:
I see the reason that "required_one_of" was specified for "any", "host" and "prefix". This is a good way of eliminating the need for the "config" code to handle the invalid case of a standalone "port" attribute.

The reason for the suggestion here was different: It was to eliminate the need for explicit handling in the "config" code for the illogical playbook case of an empty "ethertype" attribute. I suppose, however, that it might be convenient for a user to do that when deleting the an ethertype attribute.

I agree that the model is workable without explicitly requiring one of the values to be specified, so you can mark this issue "resolved".